sql/CMakeLists.txt

```diff
@@ -294,6 +294,35 @@ TARGET_LINK_LIBRARIES(sql_builtins ${MYSQLD_STATIC_PLUGIN_LIBS})
 
 MYSQL_ADD_EXECUTABLE(mysqld ${MYSQLD_SOURCE} DESTINATION ${INSTALL_SBINDIR} COMPONENT Server)
 
+IF(AIX OR CMAKE_SYSTEM_NAME STREQUAL AIX)
+  SET(SYMBOL_LIB libsql.a libpartition.a libsql_sequence.a ../mysys_ssl/libmysys_ssl.a ../mysys/libmysys.a ../dbug/libdbug.a ../strings/libstrings.a ../vio/libvio.a)
+  add_custom_target(symbols_export
+    COMMAND nm -X32_64 -e ${SYMBOL_LIB} | grep " T " | grep -v "\\." > T.exp
+    COMMAND nm -X32_64 -e ${SYMBOL_LIB} | grep " D " | grep -v "\\." > D.exp
+    COMMAND printf "{print $1}" > awk.txt
+    COMMAND awk -f awk.txt T.exp > T2.exp
+    COMMAND awk -f awk.txt D.exp > D2.exp
+    COMMAND printf "#! .\\n" > head.txt
+    COMMAND cat T2.exp D2.exp | sort | uniq > tmp.exp
+    COMMAND cat head.txt tmp.exp > ${CMAKE_SOURCE_DIR}/libmariadbd.exp
```

Comment:
should be using `CMAKE_CURRENT_BINARY_DIR` rather than the source dir.

Looks a bit verbose:
<pre>
COMMAND echo '#! .'  > ${CMAKE_CURRENT_BINARY_DIR}/libmariadb.exp
COMMAND nm -X32_64 -e  ${SYMBOL_LIB} | awk '/^[0-9]+ [DT] /{print $3}' | sort | uniq ) >> ${CMAKE_CURRENT_BINARY_DIR}/libmariadb.exp
</pre>

Reply:
I have some trouble with CMake and simple quote and space escape.
AIX nm do not have the same output than BSD or Linux, the "-B" flag permits to have a "BSD-style output".
I have **lot** of *ld: 0711-324 WARNING: Export of .FOO changed to export of FOO*. So, I erased all dot in symbol list (a symbol cannot contain a dot).
Final suggestion is
```
    COMMAND echo "#! ."  > ${CMAKE_CURRENT_BINARY_DIR}/libmariadb.exp
    COMMAND nm -B -X32_64 -e  ${SYMBOL_LIB} | awk "/^\ +[0-9]+\ [DT]\ /{print\ $3}" | sed "s|\\.||g" | sort | uniq  >> ${CMAKE_CURRENT_BINARY_DIR}/libmariadb.exp
```

Reply:
good. simpler and works was the main bit. Thanks for minimising.